src/engine/Scene.ts

```diff
@@ -85,12 +85,13 @@ export class Scene extends Class implements ICanInitialize, ICanActivate, ICanDe
   private _cancelQueue: Timer[] = [];
   private _logger: Logger = Logger.getInstance();
 
-  constructor(engine?: Engine) {
+  constructor(_engine: Engine) {
```

Comment:
Hi @dylanavery720 sorry for the very delayed response, can you make the class property private with an underscore? 

This way people won't be able to access `.engine` from a scene instance.

```typescript
export class Scene {
   private _engine: Engine;
   ...
   constructor(engine: Engine) {
      ...
      this._engine = engine
   }
```